core-libs/setup/ssr/providers/server-request-origin.ts

```diff
@@ -0,0 +1,48 @@
+import { inject } from '@angular/core';
+import { SERVER_REQUEST_ORIGIN } from '@spartacus/core';
+import { ServerOptions } from './model';
+
+/**
+ * Returns a factory function which resolves the server request origin.
+ */
+export function serverRequestOriginFactory(options?: ServerOptions): Function {
+  return (): string => {
+    const serverRequestOrigin = inject(SERVER_REQUEST_ORIGIN, {
+      optional: true,
+      skipSelf: true,
+    });
+
+    // SSR mode
+    if (serverRequestOrigin) {
+      return serverRequestOrigin;
+    }
+
+    // likely prerendering mode
+    if (options?.serverRequestOrigin) {
+      return options.serverRequestOrigin;
+    }
```

Comment:
We agreed on your solution, so we should prioritize value `options.serverRequestOrigin` over `serverRequestOrigin` injected from parent platform injector. Right?

Reply:
Correct, thanks for pointing it out. Done.